Deprecate SHA1 based algorithms for SAML signatures (#1739)
keycloak/keycloak-documentation: securing_apps/topics/saml/java/general-config/idp_element.adoc

  This is the signature algorithm that the IDP expects signed documents to use.
  Allowed values are: `RSA_SHA1`, `RSA_SHA256`, `RSA_SHA512`, and `DSA_SHA1`.
  This setting is _OPTIONAL_
  and defaults to `RSA_SHA256`. Note that `SHA1` based algorithms are deprecated and can be removed in the future.

>   It is recommended to stick to some more secure algorithm instead of `*_SHA1`. Also with `*_SHA1` algorithms verifying signatures
>   will not work if the SAML server (usually {project_name}) runs on Java 17 or later.

Reviewer: ```suggestion
 We recommend the use of some more secure algorithm instead of `*_SHA1`. Also, with `*_SHA1` algorithms, verifying signatures
  do not work if the SAML server (usually {project_name}) runs on Java 17 or higher.
```
